From d7d93a6431f0d6b41146a4e6457bea9f0208f198 Mon Sep 17 00:00:00 2001
From: Tao Zeng <tao.zeng@amlogic.com>
Date: Mon, 22 Feb 2016 14:13:44 +0800
Subject: [PATCH 5941/5965] PD#119396: cec: clock source clean up for M8baby.

From QA test report, 32768Hz clock source neither generated from
RTC oscillator nor from PWM divided from FIX PLL are both not
stable for CEC work if press TV remote key for long time. The only
way is generate clock source by ARC cpu using test mode of RTC. And
the clock source generated by ARC is 32767.7Hz measured by frequency
meter, Which is very close to 32768Hz.

This commit also fix mistakes made in:
	070a381b09e601de70117418d3c259f195c99135
	PD#93985: meson8b: improve ARC timer precision for CEC 32K

	17266b511cb2c15fe1592bbf82f04fdefeebb059
	PD#118299: cec: imporve 32k clock resolution for m8baby costdown platform

Change-Id: I1026dcab2556929894a059a8c9d17a90b731206f
---
 arch/arm/mach-meson8b/arc_trig_clk.h          | 178 +++++++++---------
 .../mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c  |  27 ---
 2 files changed, 88 insertions(+), 117 deletions(-)

diff --git a/arch/arm/mach-meson8b/arc_trig_clk.h b/arch/arm/mach-meson8b/arc_trig_clk.h
index fd6ffbb19a63..146ee2293c0f 100644
--- a/arch/arm/mach-meson8b/arc_trig_clk.h
+++ b/arch/arm/mach-meson8b/arc_trig_clk.h
@@ -1,92 +1,90 @@
 const unsigned arc_code[]={
-	0x85, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x00, 0x00, 0x00, 0x4A, 0x20, 0x00, 0x00, 0x4A, 0x21, 0x00, 0x00, 0x4A, 0x22, 0x00, 0x00,
-	0x4A, 0x23, 0x00, 0x00, 0x4A, 0x24, 0x00, 0x00, 0x4A, 0x25, 0x00, 0x00, 0x4A, 0x26, 0x00, 0x00,
-	0x4A, 0x27, 0x00, 0x00, 0x4A, 0x20, 0x00, 0x10, 0x4A, 0x21, 0x00, 0x10, 0x4A, 0x22, 0x00, 0x10,
-	0x4A, 0x23, 0x00, 0x10, 0x4A, 0x24, 0x00, 0x10, 0x4A, 0x25, 0x00, 0x10, 0x4A, 0x26, 0x00, 0x10,
-	0x4A, 0x27, 0x00, 0x10, 0x4A, 0x20, 0x00, 0x20, 0x4A, 0x21, 0x00, 0x20, 0x4A, 0x22, 0x00, 0x20,
-	0x4A, 0x23, 0x00, 0x20, 0x4A, 0x24, 0x00, 0x20, 0x4A, 0x25, 0x00, 0x20, 0x4A, 0x26, 0x00, 0x20,
-	0x4A, 0x27, 0x00, 0x20, 0x4A, 0x20, 0x00, 0x30, 0x4A, 0x21, 0x00, 0x30, 0x4A, 0x22, 0x00, 0x30,
-	0x4A, 0x23, 0x00, 0x30, 0x4A, 0x24, 0x00, 0x30, 0x0A, 0x24, 0x80, 0x3F, 0x00, 0x00, 0x00, 0x80,
-	0x4A, 0x26, 0x00, 0x70, 0x4A, 0x26, 0x00, 0x70, 0x4A, 0x20, 0x00, 0x00, 0x0A, 0x21, 0x80, 0x0F,
-	0x00, 0x00, 0x88, 0x05, 0x00, 0x19, 0x00, 0x00, 0x40, 0x21, 0x01, 0x01, 0x0C, 0x21, 0x80, 0x8F,
-	0x00, 0x00, 0x94, 0x05, 0xF0, 0x07, 0xC2, 0xFF, 0x0A, 0x23, 0x80, 0x1F, 0x10, 0xC8, 0x80, 0x00,
-	0x00, 0x1B, 0x01, 0x10, 0x0A, 0x23, 0x80, 0x1F, 0x10, 0xC8, 0x8C, 0x00, 0x00, 0x1B, 0x80, 0x1F,
-	0x00, 0x00, 0xFF, 0xFF, 0x8A, 0x23, 0x88, 0x11, 0x8A, 0x24, 0xD0, 0x12, 0x2B, 0x26, 0x00, 0x73,
-	0x00, 0x00, 0x13, 0x00, 0x2B, 0x26, 0xC0, 0x72, 0x00, 0x00, 0x00, 0x00, 0x0A, 0x22, 0x80, 0x1F,
-	0x00, 0x00, 0x3C, 0x04, 0x00, 0x1E, 0x80, 0x72, 0x00, 0x00, 0x4C, 0x00, 0x2B, 0x26, 0x00, 0x73,
-	0x00, 0x00, 0x12, 0x00, 0x2B, 0x26, 0xC0, 0x72, 0x00, 0x00, 0x01, 0x00, 0x0A, 0x22, 0x80, 0x1F,
-	0x00, 0x00, 0xFC, 0x03, 0x00, 0x1E, 0x80, 0x72, 0x00, 0x00, 0x48, 0x00, 0x69, 0x20, 0x80, 0x80,
-	0x6F, 0x26, 0x3F, 0x00, 0x4A, 0x23, 0x80, 0x12, 0x2A, 0x23, 0xC0, 0x12, 0x0A, 0x24, 0x80, 0x1F,
-	0x10, 0xC8, 0x04, 0x00, 0x00, 0x1C, 0xC0, 0x12, 0x2E, 0x09, 0x00, 0x00, 0x4A, 0x26, 0x00, 0x70,
-	0x4A, 0x26, 0x00, 0x70, 0x4A, 0x26, 0x00, 0x70, 0x6F, 0x21, 0x3F, 0x00, 0x4A, 0x26, 0x00, 0x70,
-	0x4A, 0x26, 0x00, 0x70, 0x0A, 0x24, 0x80, 0x1F, 0x00, 0x00, 0x88, 0x88, 0x0A, 0x23, 0x80, 0x1F,
-	0x10, 0xC8, 0x04, 0x00, 0x00, 0x1B, 0x00, 0x13, 0x20, 0x20, 0x80, 0x0F, 0x00, 0x00, 0xC4, 0x03,
-	0x0A, 0x24, 0x80, 0x1F, 0x00, 0x00, 0x77, 0x77, 0x0A, 0x23, 0x80, 0x1F, 0x10, 0xC8, 0x04, 0x00,
-	0x00, 0x1B, 0x00, 0x13, 0x20, 0x20, 0x80, 0x0F, 0x00, 0x00, 0xE0, 0x03, 0xFC, 0x1C, 0x08, 0xB0,
-	0xFC, 0x1C, 0x88, 0xB0, 0xFC, 0x1C, 0xC8, 0xB7, 0x4A, 0x26, 0x00, 0x70, 0x0A, 0x20, 0x80, 0x0F,
-	0x10, 0xC1, 0x44, 0x98, 0x0A, 0x22, 0x80, 0x0F, 0xFF, 0xFF, 0xFF, 0xFF, 0x00, 0x18, 0x80, 0x00,
-	0x4A, 0x26, 0x00, 0x70, 0x4A, 0x26, 0x00, 0x70, 0x4A, 0x26, 0x00, 0x70, 0x04, 0x14, 0x1F, 0x34,
-	0x04, 0x14, 0x02, 0x34, 0x04, 0x14, 0x00, 0x34, 0x6F, 0x24, 0x3F, 0x00, 0xFC, 0x1C, 0x08, 0xB0,
-	0xFC, 0x1C, 0x88, 0xB0, 0xFC, 0x1C, 0xC8, 0xB7, 0x0A, 0x22, 0x80, 0x0F, 0xFF, 0xFF, 0xFF, 0xFF,
-	0x00, 0x1E, 0x80, 0x70, 0x10, 0xC8, 0x8C, 0x00, 0x0A, 0x22, 0x80, 0x0F, 0x34, 0x12, 0xCD, 0xAB,
-	0x00, 0x1E, 0x80, 0x70, 0x10, 0xC8, 0x04, 0x00, 0x04, 0x14, 0x1F, 0x34, 0x04, 0x14, 0x02, 0x34,
-	0x04, 0x14, 0x00, 0x34, 0x6F, 0x24, 0x3F, 0x00, 0x6F, 0x26, 0x3F, 0x00, 0x6F, 0x27, 0x3F, 0x00,
-	0xE0, 0x7F, 0x11, 0x78, 0x0A, 0x25, 0x80, 0x0F, 0x10, 0xC8, 0x54, 0x00, 0x0A, 0x24, 0x80, 0x0F,
-	0x00, 0x00, 0x88, 0x05, 0x00, 0x15, 0x03, 0x00, 0x00, 0x14, 0x02, 0x00, 0x86, 0xEA, 0x00, 0x1C,
-	0xC0, 0x00, 0x21, 0x00, 0x20, 0x00, 0x60, 0x42, 0x19, 0x0A, 0xC5, 0x00, 0x42, 0x7B, 0xE9, 0x0B,
-	0x24, 0x80, 0x1A, 0x62, 0x00, 0x1E, 0x80, 0x70, 0x00, 0x00, 0x88, 0x05, 0x0B, 0xF0, 0xE0, 0x78,
-	0x42, 0x7B, 0x71, 0x7B, 0xD1, 0x0B, 0x22, 0x80, 0x1A, 0x62, 0x00, 0x1E, 0x80, 0x70, 0x00, 0x00,
-	0x88, 0x05, 0xE0, 0x7E, 0xF1, 0xC0, 0xE8, 0x1C, 0x48, 0xB3, 0x41, 0xC6, 0x42, 0xC7, 0x0C, 0x1C,
-	0x00, 0x34, 0x10, 0x1C, 0x40, 0x34, 0x14, 0x1C, 0x80, 0x34, 0xC3, 0x42, 0x10, 0xC8, 0x4C, 0x00,
-	0x60, 0x82, 0x45, 0x23, 0xC3, 0x07, 0x60, 0xA2, 0xC3, 0x42, 0xEE, 0xEE, 0xAA, 0xAA, 0x00, 0x1E,
-	0x80, 0x70, 0x10, 0xC8, 0x04, 0x00, 0xC3, 0x42, 0x10, 0xC8, 0x14, 0x00, 0x60, 0x82, 0x92, 0xBB,
-	0x60, 0xA2, 0xC3, 0x42, 0x10, 0xC8, 0x40, 0x07, 0x60, 0x82, 0x85, 0xBB, 0x60, 0xA2, 0x48, 0x45,
-	0x0A, 0x21, 0x80, 0x2F, 0x00, 0x00, 0x80, 0x09, 0x0A, 0x20, 0x80, 0x2F, 0x00, 0x00, 0x7F, 0x09,
-	0xCB, 0x47, 0x10, 0xC8, 0x04, 0x00, 0x4E, 0x70, 0x40, 0x85, 0x15, 0x0A, 0x3E, 0x01, 0x50, 0x22,
-	0x0E, 0x01, 0x46, 0x0F, 0xEF, 0xFF, 0x02, 0x40, 0x07, 0xF0, 0xE0, 0x78, 0x45, 0x22, 0x0E, 0x04,
-	0x36, 0x0F, 0xEF, 0xFF, 0x22, 0x40, 0xC0, 0xA5, 0x40, 0x87, 0xE1, 0x0A, 0x81, 0x8F, 0xDD, 0xDD,
-	0xEE, 0xEE, 0x00, 0x1F, 0x80, 0x14, 0x6F, 0x23, 0x3F, 0x00, 0x69, 0x20, 0x40, 0x00, 0x4A, 0x26,
-	0x00, 0x70, 0x4A, 0x26, 0x00, 0x70, 0x4A, 0x26, 0x00, 0x70, 0xE0, 0xF1, 0x00, 0x02, 0x01, 0x00,
-	0x1A, 0x00, 0x00, 0x00, 0xB2, 0x01, 0x00, 0x00
+	0x00000285 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x00000000 , 0x00000000 , 0x00000000,
+	0x00000000 , 0x0000204a , 0x0000214a , 0x0000224a,
+	0x0000234a , 0x0000244a , 0x0000254a , 0x0000264a,
+	0x0000274a , 0x1000204a , 0x1000214a , 0x1000224a,
+	0x1000234a , 0x1000244a , 0x1000254a , 0x1000264a,
+	0x1000274a , 0x2000204a , 0x2000214a , 0x2000224a,
+	0x2000234a , 0x2000244a , 0x2000254a , 0x2000264a,
+	0x2000274a , 0x3000204a , 0x3000214a , 0x3000224a,
+	0x3000234a , 0x3000244a , 0x3f80240a , 0x80000000,
+	0x7000264a , 0x7000264a , 0x0000204a , 0x0f80210a,
+	0x057c0000 , 0x00001900 , 0x01012140 , 0x8f80210c,
+	0x05880000 , 0xffc207f0 , 0x1f80230a , 0x0080c810,
+	0x10011b00 , 0x1f80230a , 0x008cc810 , 0x1f801b00,
+	0xffff0000 , 0x1188238a , 0x12d0248a , 0x7300262b,
+	0x00130000 , 0x72c0262b , 0x00000000 , 0x1f80220a,
+	0x043c0000 , 0x72801e00 , 0x004c0000 , 0x7300262b,
+	0x00120000 , 0x72c0262b , 0x00010000 , 0x1f80220a,
+	0x03fc0000 , 0x72801e00 , 0x00480000 , 0x80802069,
+	0x003f266f , 0x1280234a , 0x12c0232a , 0x1f80240a,
+	0x0004c810 , 0x12c01c00 , 0x0000092e , 0x7000264a,
+	0x7000264a , 0x7000264a , 0x003f216f , 0x7000264a,
+	0x7000264a , 0x1f80240a , 0x88880000 , 0x1f80230a,
+	0x0004c810 , 0x13001b00 , 0x0f802020 , 0x03c40000,
+	0x1f80240a , 0x77770000 , 0x1f80230a , 0x0004c810,
+	0x13001b00 , 0x0f802020 , 0x03e00000 , 0xb0081cfc,
+	0xb0881cfc , 0xb7c81cfc , 0x7000264a , 0x0f80200a,
+	0x9844c110 , 0x0f80220a , 0xffffffff , 0x00801800,
+	0x7000264a , 0x7000264a , 0x7000264a , 0x341f1404,
+	0x34021404 , 0x34001404 , 0x003f246f , 0xb0081cfc,
+	0xb0881cfc , 0xb7c81cfc , 0x0f80220a , 0xffffffff,
+	0x70801e00 , 0x008cc810 , 0x0f80220a , 0xabcd1234,
+	0x70801e00 , 0x0004c810 , 0x341f1404 , 0x34021404,
+	0x34001404 , 0x003f246f , 0x003f266f , 0x003f276f,
+	0x78117fe0 , 0x0f80250a , 0x0054c810 , 0x0f80240a,
+	0x057c0000 , 0x00031500 , 0x00021400 , 0x1c00ea86,
+	0x002100c0 , 0x42600020 , 0x00c50a19 , 0x0be97b42,
+	0x621a8024 , 0x70801e00 , 0x057c0000 , 0x78e0f00b,
+	0x7b717b42 , 0x80220bd1 , 0x1e00621a , 0x00007080,
+	0x7ee0057c , 0x1cecc0f1 , 0xc641b348 , 0x1c0cc742,
+	0x1c103400 , 0x42c33440 , 0x004cc810 , 0x23458260,
+	0xa26007c3 , 0xeeee42c3 , 0x1e00aaaa , 0xc8107080,
+	0x42c30004 , 0x0014c810 , 0xbb928260 , 0x42c3a260,
+	0x0740c810 , 0xbb858260 , 0x4548a260 , 0x2f80200a,
+	0x09800000 , 0xc81047cb , 0x702e0004 , 0x0a158540,
+	0x2250013e , 0x0f52010e , 0x4002ffef , 0x78e0f007,
+	0x040e2245 , 0xffef0f42 , 0xa5c04002 , 0x0ae18740,
+	0xdddd8f81 , 0x1f00eeee , 0x236f1440 , 0x2069003f,
+	0x264a0040 , 0x264a7000 , 0x264a7000 , 0xf1e07000,
+	0x00010200 , 0x0000001a , 0x000001b2 , 0x00000000
 };
-
diff --git a/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c b/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c
index ea9baa247296..74236dacbe52 100755
--- a/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c
+++ b/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c
@@ -61,32 +61,6 @@ void cec_enable_irq(void)
     hdmi_print(LOW, CEC "enable:int mask:0x%x\n", aml_read_reg32(P_AO_CEC_INTR_MASKN));
 }
 
-void gpiox_10_pin_mux_mask(void){
-    //GPIOX_10 pin mux masked expect PWM_E
-    aml_set_reg32_bits(P_PERIPHS_PIN_MUX_3, 0x0, 22, 1); //0x202f bit[22]: XTAL_32K_OUT 
-    aml_set_reg32_bits(P_PERIPHS_PIN_MUX_3, 0x0,  8, 1); //0x202f bit[ 8]: Tsin_D0_B
-    aml_set_reg32_bits(P_PERIPHS_PIN_MUX_4, 0x0, 23, 1); //0x2030 bit[23]: SPI_MOSI
-    aml_set_reg32_bits(P_PERIPHS_PIN_MUX_6, 0x0, 17, 1); //0x2032 bit[17]: UART_CTS_B
-    aml_set_reg32_bits(P_PERIPHS_PIN_MUX_7, 0x0, 31, 1); //0x2033 bit[31]: PWM_VS
-    aml_set_reg32_bits(P_PERIPHS_PIN_MUX_9, 0x1, 19, 1); //0x2035 bit[19]: PWM_E
-}
-
-void pwm_e_config(void){
-    //PWM E config
-    aml_set_reg32_bits(P_PWM_PWM_E,    0x32aa, 16, 16); //0x21b0 bit[31:16]: PWM_E_HIGH counter
-    aml_set_reg32_bits(P_PWM_PWM_E,    0x32aa,  0, 16); //0x21b0 bit[15: 0]: PWM_E_LOW counter
-    aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x1, 15,  1); //0x21b2 bit[15]   : PWM_E_CLK_EN
-    aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x0,  8,  7); //0x21b2 bit[14: 8]: PWM_E_CLK_DIV
-    aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x3,  4,  2); //0x21b2 bit[5 : 4]: PWM_E_CLK_SEL :0x2 sleect fclk_div3:850MHz
-    aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x1,  0,  1); //0x21b2 bit[0]    : PWM_E_EN
-}
-
-void pwm_out_rtc_32k(void){
-    gpiox_10_pin_mux_mask(); //enable PWM_E pin mux
-    pwm_e_config();  //PWM E config   
-    hdmi_print(INF, CEC "Set PWM_E out put RTC 32K!\n");
-}
-
 void cec_hw_reset(void)
 {
     //unsigned long data32;
@@ -488,7 +462,6 @@ void cec_polling_online_dev(int log_addr, int *bool)
 void ao_cec_init(void)
 {
     unsigned long data32;
-    pwm_out_rtc_32k();  //enable RTC 32k
     // Assert SW reset AO_CEC
     data32  = 0;
     data32 |= 0 << 1;   // [2:1]    cntl_clk: 0=Disable clk (Power-off mode); 1=Enable gated clock (Normal mode); 2=Enable free-run clk (Debug mode).
-- 
2.19.0

